src/core/service.c

```diff
@@ -990,7 +990,7 @@ static int service_load_pid_file(Service *s, bool may_warn) {
         if (r < 0)
                 return r;
 
-        r = unit_watch_pid(UNIT(s), pid);
+        r = unit_watch_pid(UNIT(s), pid, true);
```

Comment:
this should have "false" as parameter.

The thing is that it is unfortunately completely OK if the same process is associated with multiple units. this happens all the time, for example in getty-like services where the main process also allocates a scope unit. 

hence: if we just forked() off a child, then we can be sure that its PID is otherwise unused, and hence can set exclusive=true. But if we learnt about a PID in any other form (for example via PID file, via searching, MAINPID= and so on), then we should leave exclusive=false.

Reply:
I wouldn't have expectec that MAINPID= could also be used to import external processes. This sounds pretty strange. Do you have an example ?

Reply:
a real one no. a theoretic one, yes: some program that opens a scope for some purpose (and thus migrates to a different cgroup) but still wants to remain MAINPID of the originating service.